### ISSUE TITLE
Get end-to-end tests working

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#233]
### Context

We have unit tests for a lot of functionality including processors working in CI, however we don't test the system as a whole in CI.

### Problem or idea

We need tests that test a survey job queuing downloader jobs, which download the data and then queue processor jobs which process it so we know the whole flow works.

### Solution or next step

Create a single end to end test as a proof of concept. This involves creating a test Nomad client which can run alongside the normal local-development client. We'll want more than a single end-to-end test but having one will show the infrastructure is there and provide a template for others.

### New Issue Checklist

- [x] The title is short and descriptive.
- [x] You have explained the context that led you to write this issue.
- [x] You have reported a problem or idea.
- [x] You have proposed a solution or next step.

**Comment by kurtwheeler** [comment]
I will note here so that I've recorded it somewhere. I tried using Nomad's [load option](https://www.nomadproject.io/docs/drivers/docker.html#load) to avoid needing to push the image to a local registry since that's a slow step. However that option only works if you tell Nomad to download the image using the [artifact stanza](https://www.nomadproject.io/docs/job-specification/artifact.html). This doesn't help our use case because we already have the image we want Nomad to use on the disk, there just doesn't appear to be any way to have it use a local image whatsoever.
